aeon/anomaly_detection/_lof.py

```diff
@@ -0,0 +1,169 @@
+    p : integer, optional (default = 2)
+        Parameter for the Minkowski metric
+    metric_params : dict, optional (default = None)
+        Additional keyword arguments for the metric function.
+    n_jobs : int, optional (default = 1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only kneighbors and kneighbors_graph methods.
+    novelty : bool (default=False)
+        By default, LocalOutlierFactor is only meant to be used for outlier
+        detection (novelty=False). Set novelty to True if you want to use
+        LocalOutlierFactor for novelty detection. In this case be aware that
+        that you should only use predict, decision_function and score_samples
+        on new unseen data and not on the training set.
+    """
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:univariate": True,
+        "capability:missing_values": False,
+        "fit_is_empty": False,
+        "python_dependencies": ["pyod"],
+    }
+
+    def __init__(
+        self,
+        n_neighbors: int = 20,
+        algorithm: Optional[str] = "auto",
+        leaf_size: int = 30,
+        metric: str = "minkowski",
+        p: int = 2,
+        metric_params: Optional[dict] = None,
+        n_jobs: int = 1,
+        window_size: int = 10,
+        stride: int = 1,
+    ):
+        _check_soft_dependencies(*self._tags["python_dependencies"])
+        from pyod.models.lof import LOF as PyOD_LOF
+
+        # Set a default contamination value internally
+        contamination = 0.1
+
+        model = PyOD_LOF(
+            n_neighbors=n_neighbors,
+            algorithm=algorithm,
+            leaf_size=leaf_size,
+            metric=metric,
+            p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs,
+            contamination=contamination,  # Only for PyOD LOF
+            novelty=False,  # Initialize unsupervised LOF (novelty=False)
+        )
+        self.n_neighbors = n_neighbors
+        self.algorithm = algorithm
+        self.leaf_size = leaf_size
+        self.metric = metric
+        self.p = p
+        self.metric_params = metric_params
+        self.n_jobs = n_jobs
+        super().__init__(pyod_model=model, window_size=window_size, stride=stride)
+
+    def _fit(self, X: np.ndarray, y: Union[np.ndarray, None] = None) -> None:
+        # Set novelty to True for semi-supervised learning
+        self.pyod_model.novelty = True
+        super()._fit(X, y)
+
+    def _predict(self, X: np.ndarray) -> np.ndarray:
+        return super()._predict(X)
+
+    def _fit_predict(
+        self, X: np.ndarray, y: Union[np.ndarray, None] = None
+    ) -> np.ndarray:
+        # Set novelty to False for unsupervised learning
+        self.pyod_model.novelty = False
+        return super()._fit_predict(X, y)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
```

Comment:
This is responsible for the test failures:

```suggestion
    @classmethod
    def _get_test_params(cls, parameter_set="default"):
```